p2p/conn/secret_connection.go

```diff
@@ -267,7 +267,10 @@ func genChallenge(loPubKey, hiPubKey *[32]byte) (challenge *[32]byte) {
 }
 
 func signChallenge(challenge *[32]byte, locPrivKey crypto.PrivKey) (signature crypto.Signature) {
-	signature = locPrivKey.Sign(challenge[:])
+	signature, err := locPrivKey.Sign(challenge[:])
+	if err != nil {
+		panic(err)
```

Comment:
can you add a todo to modify func to return err?

Reply:
Sure! Will do (and also open an issue so that doesn't get lost)